### PR TITLE
Improve HANA cli installation when BACKEND is not qemu

### PR DESCRIPTION
**Pull request by alvarocarvajald**
For the HANA cli installation in qemu, we expect the job to have correctly configured settings so a second disk with enough space for HANA is configured in the VM. When backend is not qemu, the test will rely on whatever disks are already configured in the system, be it bare metal or a ppc64le LPAR; in these cases, the tests selects `/dev/sdb` by default, even if this device does not have enough space for HANA or if it does not exist.

This commit introduces changes to make the `sles4sap/hana_install` test module select a device for HANA when the backend is not qemu. This is done by probing for available block devices, and verifying if these have enough space for HANA as calculated from the system RAM. If no disks are found with enough space for HANA, test will fail.

Since SLES4SAP is installed by default with LVM for swap and root FS, the test module will filter out block devices that are configured as LVM physical volumes. It will also check for multipath devices if the installation was done with multipath configured. This is determined by the `MULTIPATH=1` and `MULTIPATH_CONFIRM=yes` settings.

Test module depends on the output of the `lsblk` command to probe devices and get their size.

- Related ticket: N/A
- Needles: N/A
- Verification run: [on ppc64le with multipath](http://mango.suse.de/tests/2955), [on ppc64le without multipath](http://mango.suse.de/tests/2954), [on x86_64 over qemu](http://mango.suse.de/tests/2941)

### CODE REVIEW

**Comment by alvarocarvajald** [comment]
Found a way to drop some lines in the module. Will test and then push changes.

File: tests/sles4sap/hana_install.pm [hunk 1]
```diff
@@ -65,10 +110,15 @@ sub run {
         # in a different backend, assume sdb exists. Always create mountpoints.
         foreach (keys %mountpts) { assert_script_run "mkdir -p $mountpts{$_}->{mountpt}"; }
         if ((check_var('BACKEND', 'qemu') and get_var('HDDSIZEGB_2')) or !check_var('BACKEND', 'qemu')) {
-            my $device = (check_var('HDDMODEL', 'scsi-hd') or !check_var('BACKEND', 'qemu')) ? '/dev/sdb' : '/dev/vdb';
-            script_run "wipefs -f $device; wipefs -f ${device}1";
+            my $device = check_var('HDDMODEL', 'scsi-hd') ? '/dev/sdb' : '/dev/vdb';
+            # We need 2.5 times $RAM + 50G for HANA installation.
+            # On qemu, we assume 2nd disk was properly configured in settings
+            # On BACKENDS different than qemu make sure we get a device with at least that size
+            $device = $self->get_hana_device_from_system(($RAM * 2.5) + 50000) unless check_var('BACKEND', 'qemu');
```
Comment (ldevulder) [inline tests/sles4sap/hana_install.pm:117]: This command should work on qemu backend also, no? If yes maybe it's easier to execute it in any case? The check with the `HDDMODEL` could be removed also in that case. WDYT?

Comment (alvarocarvajald) [reply tests/sles4sap/hana_install.pm:117]: Let's test it on qemu. On it!

**Comment by alvarocarvajald** [comment]
Verification runs with latest changes:

- x86_64 on qemu: http://mango.suse.de/tests/2964
- ppc64le LPAR with multipath: http://mango.suse.de/tests/2963
- ppc64le LPAR without multipath: http://mango.suse.de/tests/2962
- ppc64le on qemu: TBA

File: tests/sles4sap/hana_install.pm (hunk 1 above)
Comment (alvarocarvajald) [reply tests/sles4sap/hana_install.pm]: It works on qemu. I added a verification run below, and will also add a verification run for PowerKVM. Code has been pushed if you want to review again.

**Comment by alvarocarvajald** [comment]
Disk size conversion in this PR (bytes to Mbytes) is wrong. Bytes should be divided by 1024 twice to take the number to Mbytes. PR #10653 fixes this.

**Pull request merged**